docs/contracts/v4/guides/15-single-sided-liquidity.mdx

```diff
@@ -0,0 +1,423 @@
+        uint160 targetSqrtPriceX96 = uint160((uint256(sqrtPriceX96) * 974679) / 1_000_000);
+        int24 targetTick = TickMath.getTickAtSqrtPrice(targetSqrtPriceX96);
+        int24 alignedTick = (targetTick / TICK_SPACING) * TICK_SPACING;
+
+        int24 tickUpper = alignedTick;
+        int24 tickLower = tickUpper - TICK_SPACING;
+
+        if (tickLower <= currentTick) {
+            tickLower += TICK_SPACING;
+            tickUpper += TICK_SPACING;
+        }
+        
+        // Calculate liquidity from token1 (USDC)
+        uint128 liquidity = LiquidityAmounts.getLiquidityForAmount1(
+            TickMath.getSqrtPriceAtTick(tickLower),
+            TickMath.getSqrtPriceAtTick(tickUpper),
+            token1Amount
+        );
+
+        uint256 requiredToken1 = LiquidityAmounts.getAmount1ForLiquidity(
+        TickMath.getSqrtPriceAtTick(tickLower),
+        TickMath.getSqrtPriceAtTick(tickUpper),
+        liquidity
+        );
+        
+        uint256 amount1Max = requiredToken1 + 1_000; // 0.001 USDC buffer
+        uint256 amount0Max = 1e14; // small ETH buffer
+        bytes memory hookData = new bytes(0);
+
+        //Token Approval
+        vm.startBroadcast();
+        if (!currency1.isAddressZero()) {
+            IERC20(Currency.unwrap(currency1)).approve(address(PERMIT2), type(uint256).max);
+            PERMIT2.approve(Currency.unwrap(currency1), address(posm), type(uint160).max, type(uint48).max);
+        }
+        vm.stopBroadcast();
+
+        //Minting a Position
+        vm.startBroadcast();
+        IPositionManager(address(posm)).mint(
+            pool, tickLower, tickUpper, liquidity, amount0Max, amount1Max, msg.sender, block.timestamp + 60, hookData
+        );
+        vm.stopBroadcast();
+
+    }
+}
+```
+
+### Getting the tokenId
+
+Now that we have minted a new position, we want to read the response to our `Mint` function call to get **the token id**.
+
+We will need the tokenId to fetch the Position Info and use it to burn and take out position and can use the nextTokenId function from Position Manager contract.
+
+```solidity
+uint256 tokenId = posm.nextTokenId() - 1;
+```
+
+We have created our Range Order Position, now we need to monitor it and close the position when desired tick range is crossed and token1 increases in price value compare to token0.
+
+## Closing the Limit Order
+
+We have to call the modifyLiquidities function of Position Manager contract in order to burn a position and remove our token.
+
+Uniswap v4 allows us to use Actions and encode multiple actions to be sent as params. For ex: 0x030x0e means burn a position and take the tokens.
+
+And for that, we initialize the burnParams:
+```solidity
+bytes memory burnParams = abi.encode(
+    tokenId,
+    uint128(0),
+    uint128(0),
+    hookData
+);
+```
+
+And then, initialize the takeParams:
+```solidity
+bytes memory takeParams = abi.encode(
+    currency1,
+    msg.sender,
+    uint256(0)
+);
+```
+
+## Burn and Take
+
```

Comment:
again im not sure we want full code snippets here; we should probably just reference the burn guide